### PR TITLE
Support of `./extensions/` directory in template

### PR DESCRIPTION
**Pull request by con-f-use**
In `cookiecutter.json`, you can now use the `_extensions` keyword to import extensions for a new `./extensions` directory in the template (similar to `./hooks/`).

image

For some related discussion see e.g.: #1211 and #547

The [updated documentation](https://github.com/con-f-use/cookiecutter/blob/2bbea175558bf4b7fdb83630ec0d08c029d840b1/docs/advanced/template_extensions.rst) has a more detailed description.

I would be willing to write additional tests to test this functionality.

Also. the "new in version ???" from my documentation should be updated before merging and releasing, same as release notes obviously.

### CODE REVIEW

**Comment by con-f-use** [comment]
As mentioned, this is similar to #1240 and #944 .

@ssbarnea @insspb 
I'm biased, but I'd gravitate toward my more simple solution that involves less code. For the same reason I'd also vote for using the _extensions keyword in the json for both local and installed extensions.

If anything, make the folder name in #1240 just extensions instead of local_extensions to go along with ./hooks/.

File: cookiecutter/config.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def merge_configs(default, overwrite):
 def get_config(config_path):
     """Retrieve the config from the specified path, returning a config dict."""
     if not os.path.exists(config_path):
-        raise ConfigDoesNotExistException
+        raise ConfigDoesNotExistException(config_path)
```
Comment (insspb) [inline cookiecutter/config.py:65]: Out of this issue and already changed.

File: docs/advanced/template_extensions.rst [hunk 2]
```diff
@@ -15,14 +15,34 @@ To do so, a template author must specify the required extensions in ``cookiecutt
...
+        "_extensions": ["jinja2_time.TimeExtension", "myext.RegexReplace"]
     }
 
 On invocation Cookiecutter tries to import the extensions and add them to its environment respectively.
 
 In the above example, Cookiecutter provides the additional tag `now`_, after
 installing the `jinja2_time.TimeExtension`_ and enabling it in ``cookiecutter.json``.
 
+You can place small custom extensions in your cookiecutter's ``./extensions`` directory
+(*New in Cookiecutter ???*).
```
Comment (insspb) [inline docs/advanced/template_extensions.rst:27]: Please remove this line.

File: docs/advanced/template_extensions.rst [hunk 3]
```diff
@@ -15,14 +15,34 @@ To do so, a template author must specify the required extensions in ``cookiecutt
...
-        "_extensions": ["jinja2_time.TimeExtension"]
+        "_extensions": ["jinja2_time.TimeExtension", "myext.RegexReplace"]
     }
 
 On invocation Cookiecutter tries to import the extensions and add them to its environment respectively.
 
 In the above example, Cookiecutter provides the additional tag `now`_, after
 installing the `jinja2_time.TimeExtension`_ and enabling it in ``cookiecutter.json``.
 
+You can place small custom extensions in your cookiecutter's ``./extensions`` directory
```
Comment (insspb) [inline docs/advanced/template_extensions.rst:26]:
```suggestion
You can place custom Jinja2 extensions in your cookiecutter's ``./extensions`` directory
```

File: docs/advanced/template_extensions.rst (hunk 2 above)
Comment (insspb) [reply docs/advanced/template_extensions.rst:27]: (*New in Cookiecutter ???*).

File: cookiecutter/config.py (hunk 1 above)
Comment (con-f-use) [reply cookiecutter/config.py:65]: Yeah, it was useful for debugging during development. Can I just leave it in or does it conflict?

**Comment by insspb** [comment]
@con-f-use  As I mention earlier some tests required. 
At least these behavior cases should be covered:
User requested an custom extension, extension exist and work.
User requested an custom extension, but extension not exist in `extension` folder files.
User requested an custom extension, file/class exist in folder, but is nor an extension.

This feature can be useful for some part of users, who want to deliver own extensions without pip packages. But it is not just path extension.

**Comment by con-f-use** [comment]
> User requested an custom extension, but extension not exist in `extension` folder files.
> User requested an custom extension, file/class exist in folder, but is nor an extension

I wouldn't worry about these two cases. The python import system will catch  the first one. 

For the second IMHO it is not necessary to check if it is an extension. If Jinja cannot find an extention, it will complain. If it is a python module or package instead of an extension than I'd not explicitly forbid that.

**Comment by insspb** [comment]
@con-f-use I just found #1240 and seems it is more correct implementation of same thing and almost done. 
I will close this one and #944  as duplicates of #1240

**Pull request closed without merging**